src/components/Task/MassActions/ConfirmationModal/MassActionsTasksConfirmationModal.tsx

```diff
@@ -20,23 +27,71 @@ export const MassActionsTasksConfirmationModal: React.FC<
 > = ({ open, idsCount, action, setOpen, onConfirm }) => {
   const { t } = useTranslation();
 
+  // KG: useState for checkbox
+  const [checked, setChecked] = useState(false);
+
+  const shouldConfirmDeletion = action === 'delete' && idsCount > 5;
```

Comment:
Every time the component rerenders, it is recalculating this variable. In this case, it's a pretty simple equation, but if we wanted to Optimize this component, we could wrap it in a useMemo React hook.
https://react.dev/reference/react/useMemo

```suggestion
  const shouldConfirmDeletion = useMemo(() => (
    action === 'delete' && idsCount > 5
  ), [action, idsCount]);
```